### PR TITLE
Gemini request mech tool

### PR DESCRIPTION
**Pull request by victorpolisetty**
## Proposed changes

Adding Google Gemini as a mech tool. To be utilized as an alternative A.I. LLM tool to answer questions.

Would appreciate review as this is my first mech tool attempt. @0xArdi @dvilelaf 

## Fixes

NA.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [x] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works

## Further comments

Currently only supporting single question answers (no memory) and no picture to text models (vision-pro models).

### CODE REVIEW

File: packages/victorpolisetty/customs/gemini_request/gemini_request.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    response = engine.generate_content(
+        prompt,
+        generation_config=genai.types.GenerationConfig(
+            candidate_count=candidate_count,
+            stop_sequences= stop_sequences,
+            max_output_tokens=max_output_tokens,
+            temperature=temperature)
+    )
+
+    return response.text, prompt, None, counter_callback
```
Comment (dvilelaf) [inline packages/victorpolisetty/customs/gemini_request/gemini_request.py:71]: Does response always contain a .text property, even in error cases? If not, we might need to check before returning. cc @0xArdi

**Review by dvilelaf — APPROVED** [review]

**Comment by 0xArdi** [comment]
@victorpolisetty looks good! We need to add a gemini key to our CI so you can add tests to make sure everything is OK. I will ping here once we do that.

**Comment by victorpolisetty** [comment]
> @victorpolisetty looks good! We need to add a gemini key to our CI so you can add tests to make sure everything is OK. I will ping here once we do that.

**Comment by 0xArdi** [comment]
@victorpolisetty can you resolve the merge conflicts?

**Comment by victorpolisetty** [comment]
> @victorpolisetty can you resolve the merge conflicts?

Thanks for the review. Will take a look this weekend :)

**Comment by victorpolisetty** [comment]
@0xArdi Fixed merge conflicts

**Comment by victorpolisetty** [comment]
@0xArdi Added deps. Please advise on further steps.

**Pull request merged**